### ISSUE TITLE
updating to 1.2.18 breaks audio on asahi linux (apple silicon)

### ISSUE THREAD

**Issue by dkwo** [lsp-plugins/lsp-plugins#463]
Updating from 1.2.14 to 1.2.19 breaks audio for me, on asahi linux.
The wireplumber now logs that `control 'cbe_0' can not be set: No such file or directory`.
Any hint of what could be done to address this?

The asahi-audio package https://github.com/AsahiLinux/asahi-audio makes use of lsp-plugins to control speakers.
Ref: https://github.com/AsahiLinux/asahi-audio/issues/48

**Comment by mkurz** [comment]
For the record, not just 1.2.14 works, but also 1.2.17 is still working. 1.2.19 breaks for me too.

**Comment by dkwo** [comment]
Indeed, the breaking happens in going from 1.2.17 (working) to 1.2.18 (not working) for me as well.

**Comment by mkurz** [comment]
(I did not test 1.2.18, only 1.2.19)

**Comment by sadko4u** [comment]
Hello! What plugin you're talking about?

**Comment by sadko4u** [comment]
I've checked the LV2 TTL files for 1.2.13 and 1.2.14 builds and don't see any plugin that introduces the 'cbe_0' parameter.

For 1.2.13 the command gives nothing:
```
1.2.13/Linux-x86_64/lsp-plugins-lv2-1.2.13-Linux-x86_64/usr/lib/lv2/lsp-plugins.lv2> grep -R cbe_0 *
```

Also this command for 1.2.17 gives nothing:
```
1.2.17/lsp-plugins-1.2.17-Linux-x86_64/LV2/lsp-plugins.lv2> grep -R cbe_0 *
```

And for 1.2.19 gives nothing:
```
1.2.19/lsp-plugins-1.2.19-Linux-x86_64/LV2/lsp-plugins.lv2> grep -R cbe_0 *
```

It seems that software tries to write non-existing plugin parameter. Maybe the library that allows to interact with plugins became more strict?

**Comment by sadko4u** [comment]
The same is true for 1.2.14, no mention of `cbe_0` parameter:
```
1.2.14/Linux-x86_64/lsp-plugins-lv2-1.2.14-Linux-x86_64/usr/lib/lv2/lsp-plugins.lv2> grep -R cbe_0 *
```

**Comment by sadko4u** [comment]
If we talk about `cbe_X` parameters, it seems that code is somehow utilizing multiband tools like compressor. For compressor, [here is the diff of the LV2 TTL file between 1.2.17 and 1.2.19 versions](https://github.com/user-attachments/files/17691029/mb_compressor.txt).
If we look at it, here is the summary:
- added `sce_0` - `sce_7` legacy LV2 parameters to control sidechain source for compressor;
- added the `link` parameter which is `patch:writable`, not a legacy optional LV2 parameter;
- `portIndex` for parameters have been changed because of adding new legacy LV2 parameters.

I suppose there can be the following reasons:
- the plugin hosting sofware has problems dealing with the `link` parameter;
- there is something going wrong with shared memory links which have been added to LSP Plugins in 1.2.18.

Anyway, need more diagnostics information.

**Comment by chadmed** [comment]
I cannot reproduce this at all on any of my machines. 1.2.19 works totally fine here with asahi-audio versions 2.3 and 2.4 on pipewire 1.2.6 and wireplumber 0.5.6.

**Comment by mkurz** [comment]
Like comment in the asahi issue: Patch works for me too on Asahi Linux / aarch64. Thanks!

**Comment by dkwo** [comment]
Works for me too, thanks a lot!

**Comment by sadko4u** [comment]
Fixed in 1.2.20!
